### DISCUSSION TITLE
Edited iPhone cinematic videos export with a different datetime than the original video

### DISCUSSION THREAD

**Discussion by philc** [RhetTbull/osxphotos#1589] [General]
I'm using osxphotos to export, edit, and reimport photos back into Apple Photos. When reimported, the "edited" cinematic videos appear in the wrong place in the Photos timeline view, shifted by the timezone offset difference between my Mac's local time vs. where I shot the video.

This looks like an underlying export bug in Apple Photos, which osxphotos preserves, and thought I'd post what I've found, in case someone has further context.

I've observed this issue with all cinematic videos shot with an iPhone, but not regular videos, and not regular/portrait photos.

Using the Apple Photos UI, when exporting the original video, deleting it from Photos, and reimporting it, it appears in the correct position on the timeline.

When using the "plain" export in the Photos UI (i.e. edits included), after re-importing the video, it's time-shifted (3 hours, in my case).

This also happens when exporting and reimporting the video using osxphotos. I used this command:

`osxphotos --download-missing --touch-file --selected the-output-dir`

Using exiftool, I diffed and compared the timestamp fields of the original vs the _edited version of the cinematic, some of which were different.

I set each field to match, manually, and diffed the results to compare:

    exiftool -tagsFromFile orig.mov -CreateDate -ModifyDate \
    -MediaCreateDate -MediaModifyDate -TrackCreateDate -TrackModifyDate edited.mov

    diff <(exiftool -time:all orig.mov | sort) <(exiftool -time:all edited.mov | sort)
    # (they match)

And yet, after this, the edited video still imports in the wrong place in the timeline.

HOWEVER, this does resolve the issue, and the edited video will now import in the correct place in the timeline:

    exiftool -tagsFromFile orig.mov -time:all edited.mov

`time:all`'s behavior shouldn't be different than what I did in my tag by tag version of the command, and the diff results are the same, and yet it's modified some tag which Photos uses.

I thought I'd post this workaround/resolution for anyone who encounters the same problem. Unfortunately, I still don't understand the solution. Which metadata tag, precisely, is Photos using for the file's timestamp, and why is the tag's value different for the "edited" cinematic video than the original?

**Comment by oPromessa** [comment]
Hello @philc 

You probably already found it but take a look at [Movie dates and Photos.app](https://discussions.apple.com/docs/DOC-250002750)

Didn't test/use cinematic but mov and mp4 to e date field Photos prefers to use is Keys -- which I did not find in your report. Don't your files have it?

> osxphotos --download-missing --touch-file --selected the-output-dir

To ensure all metadata is exported by osxphotos you'd have to also use --exiftool. You can also use --sidecar json to save metadata in a sidecar file which you can then use glob import (when you reimport the file **but you must keep the same name on the files edited as the original on reimport**

On import you can also use the new functionality with --exportdb to re-set metadata and album where the exported file was originally.

**Comment by RhetTbull** [comment]
@philc I believe the difference between `time:all` and your command is that `time:all` is setting the time zone offset (`OffsetTimeOriginal`) which is required to set the time zone correctly. If your image doesn't have a time zone offset (as measured from UTC) then Photos uses the computers local timezone.

You can bulk fix these using `osxphotos timewarp` or re-import using `osxphotos import` with either `--sidecar` or `--exportdb` as @oPromessa suggested. I do plan to eventually add a tool to osxphotos to facilitate this kind of work (export, edit, replace original with edited version while maintaining metadata).  See #909.

**Comment by philc** [comment]
Hey folks, thank you for the suggestions.

@oPromessa Adding `--exiftool` to `osxphotos` invocation does resolve this issue I've described with cinematic videos.

@RhetTbull I suspected something like that, but the `OffsetTimeOriginal` tag was not set on the original video, nor is it set after exporting with the `--exiftool` option.
